e2e/test/E2ETestBase.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Threading;
+using Microsoft.ApplicationInsights.DataContracts;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Microsoft.Azure.Devices.E2ETests
+{
+    /// <summary>
+    /// This class creates an instance of the logger for each test and logs the test result along with other useful information.
```

Comment:
One issue I had faced in the past, in parallelizing EventSource trace generation, is that the SDK writes all of it's events to the same EventSource providers (for tests running in parallel). The test logging, in that case, was unable to differentiate between the SDK log generated for test 1 vs test 2.

If I understand correctly, since we can now forward these traces to application insights, we should be able to add additional identifiers to each of the log statements being forwarded, and that would help us filter out SDK logs for test 1 vs SDK logs for test 2. Is that correct?

Reply:
This is, of course, with the assumption that test 1 and test 2 are both running in parallel, under the same process.

Reply:
yes we are doing that in AI right now. I'm not very sure how EventSource works.

Reply:
FYI - These additional things are being added as properties, not in the message itself. It just makes querying easier. But we can always add all that information to the message as well.

Reply:
![image](https://user-images.githubusercontent.com/14096264/87818612-b0902700-c81f-11ea-959f-f17c91bfde10.png)